File: app/src/main/java/com/shindra/LinesViewHolder.java
package com.shindra;

import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.shindra.ctslibrary.bo.Line;


public class LinesViewHolder extends RecyclerView.ViewHolder {

    ImageView mImageView;
    Button schedule_button;

    public LinesViewHolder(@NonNull View itemView) {
        super(itemView);

        mImageView = itemView.findViewById(R.id.tramLine);

        this.schedule_button = itemView.findViewById(R.id.schedule_button);
    }




    public void onBind(Line line, RecyclerItemClick callback){
        //lineName.setText(line.na);

        schedule_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                callback.onScheduleClick(line);
            }
        });
    }

}

File: app/src/main/java/com/shindra/Lines/LinesViewHolder.kt
package com.shindra.Lines

import android.view.View
import android.widget.Button
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.shindra.R
import com.shindra.ScheduleClick
import com.shindra.ctslibrary.bo.Line

class LinesViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private var mImageView: ImageView = itemView.findViewById(R.id.tramLine)
    private var scheduleButton = itemView.findViewById<Button>(R.id.schedule_button)

    fun onBind(line: Line, callback: ScheduleClick) {
        scheduleButton.setOnClickListener { callback.onScheduleClick(line) }

        when (line.name) {
            itemView.context.getString(R.string.line_id_A) -> mImageView.setImageResource(R.drawable.tram_a)
            itemView.context.getString(R.string.line_id_B) -> mImageView.setImageResource(R.drawable.tram_b)
            itemView.context.getString(R.string.line_id_C) -> mImageView.setImageResource(R.drawable.tram_c)
            itemView.context.getString(R.string.line_id_D) -> mImageView.setImageResource(R.drawable.tram_d)
            itemView.context.getString(R.string.line_id_E) -> mImageView.setImageResource(R.drawable.tram_e)
            itemView.context.getString(R.string.line_id_F) -> mImageView.setImageResource(R.drawable.tram_f)
            else -> mImageView.setImageResource(R.drawable.tram)
        }
    }

}